### PR TITLE
Update docker documentation

### PR DESCRIPTION
**Pull request by michaelkaye**
Add notes on the new Docker file in the readme, as well as clarify some things about secrets brought up by https://github.com/matrix-org/synapse/pull/3217

### CODE REVIEW

File: contrib/docker/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Synapse Docker
 
-This Docker image will run Synapse as a single process. It does not provide any
+The `matrixdotorg/synapse` Docker image will run Synapse as a single process. It does not provide any
```
Comment (jcgruenhage) [inline contrib/docker/README.md]: `docker.io/matrixdotorg/synapse`?

**Comment by erikjohnston** [comment]
Might be worth adding a note that we expect people to use docker.io in production, rather than using it from a git checkout or whatever

File: README.rst [hunk 2]
```diff
@@ -157,8 +157,9 @@ if you prefer.
 
 In case of problems, please see the _`Troubleshooting` section below.
 
-Alternatively, Andreas Peters (previously Silvio Fricke) has contributed a Dockerfile to automate the
-above in Docker at https://hub.docker.com/r/avhost/docker-matrix/tags/
+There is an offical synapse image available at https://hub.docker.com/r/matrixdotorg/synapse/tags/ which can be used with the docker-compose file available at `contrib/docker`. Further information on this including configuration options is available in `contrib/docker/README.md`.
```
Comment (richvdh) [inline README.rst]: please please please can we wrap at 80 chars!

File: contrib/docker/README.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
 # Synapse Docker
 
-This Docker image will run Synapse as a single process. It does not provide any
-database server or TURN server that you should run separately.
+The `matrixdotorg/synapse` Docker image will run Synapse as a single process. It does not provide a
```
Comment (richvdh) [inline contrib/docker/README.md]: wrapping

File: contrib/docker/README.md [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
+database server or a TURN server, you should run these separately.
```
Comment (richvdh) [inline contrib/docker/README.md]: s/,/;/ or better yet rephrase it

**Pull request merged**